### PR TITLE
Add support for aliases to `ov`

### LINKED ISSUE #4407: Add support for aliases in `ov`
## Problem

We tried adding aliases in https://github.com/WordPress/openverse/pull/4343 using associative bash arrays. However, macOS's bash version is stuck on an older version that doesn't support associative arrays. So I removed that from the PR and decided to try it a different way in a future PR as a separate issue.

## Description

Create some method of expanding aliases when running `ov`. For example, `ov j` should expand to `ov just`.

It should be possible with one-dimensional arrays, and treating odd-or-0 index as the alias and the following even index as the expansion. This would be preferred, as it's probably less overhead than something else (like the Python script).

Another option I thought of was to use Python to process argv in the entrypoint and expand the aliases there. The entrypoint could end with `python ./docker/dev_env/expand_aliases.py "$@"`, and that script could import `$OPENVERSE_PROJECT/.ov_aliases.py` using a method like https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly, and then do something like:

```py
#... import personal_aliases from $OPENVERSE_PROJECT/.ov_aliases.py

shared_aliases = {
    "j": "just",
}

all_aliases = shared_aliases | personal_aliases

args = sys.argv[1:]
if args[0] in all_aliases:
  args[0] = all_aliases[args[0]]

sys.execv(args)
```

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #4407  by @sarayourfriend 

cc @AetherUnbound and @dhruvkb who are probably the most interested with this.

## Description

An alternative to https://github.com/WordPress/openverse/pull/4408 using a Python-based approach.

This approach has all the features of the bash approach in https://github.com/WordPress/openverse/pull/4408, but none of the disadvantages. It shares the same overall problems like the aliases only being available in the entrypoint. They aren't available inside `just` recipes themselves, for example. I think that's fine though!

## Testing Instructions

Try `ov j`, `ov nuxt`, and `ov welcome`. See if it works and makes sense!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (`just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: ov [hunk 1]
```diff
@@ -57,7 +57,7 @@ fi
 case "$_cmd" in
 init)
   "$_self" build
-  "$_self" just install-hooks
+  "$_self" 'just install-hooks && just init-ov-aliases'
```
[the remark this answers is not in the document]
Comment (sarayourfriend) [reply ov:60]: Oops, yes it will, thanks!

**Comment by sarayourfriend** [comment]
> I also needed to run ov build && ov init first

Ah, of course.

Hopefully the entrypoint doesn't need to change that often, but if we find it does for some reason, we can change it to a stub script that just passes through to a mounted script that doesn't need to be copied into the container. Something like

```sh
# entrypoint.sh

"$OPENVERSE_PROJECT"/docker/dev_env/real_entrypoint.sh "$@"
```

Where `real_entrypoint.sh` has everything that's going on here. It's another layer of indirection, which is kind of annoying, but would make iteration easier.

File: docker/dev_env/expand_aliases.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import json
+import os
+import sys
+from pathlib import Path
+
+
+SHARED_ALIASES = {
+    "j": ["just"],
+    "nuxt": ["j", "p", "frontend", "dev"],
```
Comment (dhruvkb) [inline docker/dev_env/expand_aliases.py:9]: This feels like preferential treatment for the frontend 😄.

**Review by dhruvkb — APPROVED** [review]
LGTM. I would have preferred a more _standard_ way to define aliases than using a Python script, but this works well so it's okay with me.

File: docker/dev_env/expand_aliases.py (hunk 2 above)
Comment (sarayourfriend) [reply docker/dev_env/expand_aliases.py:9]: It was just the first useful one I could think of that clearly illustrated multi-term expansion :sweat_smile:

**Comment by dhruvkb** [comment]
@krysal you have to link the `ov` script to a directory that's on your `$PATH`. On a Mac, `/usr/local/bin` is typically on the `$PATH` so this should work.

```bash
sudo ln -s <path to repo>/ov /usr/local/bin/ov
```

**Comment by sarayourfriend** [comment]
Oh, that's no good! The hooks shouldn't rely on `ov` being in the path, that's definitely a bug. They should reference the script directly, I guess by relative path? I'm not sure what the CWD of a git hook is but I'm sure it's easy to find out. Issue here to fix that :https://github.com/WordPress/openverse/issues/4422

**Pull request merged**